crates/context/src/lib.rs

```diff
@@ -307,11 +307,18 @@ impl ContextManager {
         blob_id: calimero_primitives::blobs::BlobId,
         source: calimero_primitives::application::ApplicationSource,
         version: Option<semver::Version>,
+        contract_app_id: Option<String>,
     ) -> eyre::Result<calimero_primitives::application::ApplicationId> {
+        let boxed_id: Box<str> = match contract_app_id {
+            Some(s) => s.into_boxed_str(),
+            None => Box::from(""),
+        };
+
         let application = calimero_store::types::ApplicationMeta {
             blob: calimero_store::key::BlobMeta::new(blob_id),
             version: version.map(|v| v.to_string().into_boxed_str()),
             source: source.to_string().into_boxed_str(),
+            contract_app_id: Some(boxed_id),
```

Comment:
what is diff between app_id and contract_app_id?

Reply:
Can you please add a field description to the code as well so we have it documented?